unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Tests/BIWSaveControllerShould.cs

```diff
@@ -15,8 +16,8 @@ protected override IEnumerator SetUp()
         yield return base.SetUp();
         gameObject = new GameObject();
 
+        builderInWorldBridge = InitialSceneReferences.i.builderInWorldBridge;
```

Comment:
I guess this `InitialSceneReferences.i.builderInWorldBridge` should be loaded with the correct references in the test scene, right?

Following with the previous suggestion about to inject the `InitialSceneReferences` as reference in each BIW controller, I think doing this we could make these kind of tests independent of any instanced game object in the scene and we could use mocked specific instances only for testing.

Reply:
Yes, it have the correct reference in the test scene. 

Anyway, you still need to instanciate the gameObjects since this class reference to monobehaviours. If brian says that we can go for the dependency injection, I will change it!

Reply:
Remember that right now we are trying to get rid of `IntegrationTestSuite_Legacy` and this means we are going to get rid of the test scene eventually. This said, if we go for the dependency injection we eventually could load the references without needing the scene, so this would help to get to that point. What do you think?

Reply:
Sure! Lets change it then as a dependency injection!